devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,360 @@
+	log.info "zwaveEvent SecurityMessageEncapsulation called: ${cmd}, ${endpoint}"
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulatedCommand from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, endpoint = null) {
+	log.info "zwaveEvent MultiChannelCmdEncap called: '${cmd}'  endpoint '${endpoint}'"
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	log.debug "MultiChannelCmdEncap: encapsulatedCommand '${encapsulatedCommand}'"
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd, endpoint = null) {
+	log.info "zwaveEvent NotificationReport called: cmd: '${cmd}' notificationType: '${cmd.notificationType}', event: '${cmd.event}', endpoint '${endpoint}'"
+	def result = []
+
+	if (cmd.notificationType == 0x08) {
+		def value = cmd.event== 0x03? "on" : "off"
+		endpoint ? result = changeSwitch(endpoint, value) : []
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	log.info "zwaveEvent Crc16Encap called: cmd '${cmd}'"
+	def versions = commandClassVersions
+	def version = versions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+        log.warn "Unable to extract CRC16 command from $cmd"
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	log.info "zwaveEvent SensorMultilevelReport called, ${cmd}"
+	def map = [:]
+	def result = []
+	switch (cmd.sensorType) {
+		case 1:
+			map.name = "temperature"
+			def cmdScale = cmd.scale == 1 ? "F" : "C"
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+			map.unit = getTemperatureScale()
+			break		
+		case 5:
+			map.name = "humidity"
+			map.value = cmd.scaledSensorValue.toInteger()
+			map.unit = "%"
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	log.debug "SensorMultilevelReport, ${map}, ${map.name}, ${map.value}, ${map.unit}"
+	result << createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, endpoint = null) {
+	log.info "zwaveEvent ***** Unhandled Command called, cmd '${cmd}', endpoint '${endpoint}' *****"
+	[descriptionText: "Unhandled $device.displayName: $cmd", isStateChange: true]
+}
+
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping(endpoint = null) {
+	log.info "ping called: endpoint '${endpoint}' state '${state}'"
+	log.debug "pint : device.currentValue : " + device.currentValue("DeviceWatch-DeviceStatus")
```

Comment:
Since you're no longer tracking device health for the children, this should be refactored, since you will never call `ping` with an endpoint specified.

Reply:
So `ping` and `refresh` should probably query each endpoint as well as the temperature and humidity, so that `ping` just becomes a call to `refresh`